frontend/src/components/collab/chat/ChatSpace.tsx

```diff
@@ -1,155 +1,238 @@
 "use client";
 import { useEffect, useRef, useState, useLayoutEffect } from "react";
 import ChatBubble from "./ChatBubble";
-import { Button, Divider, Tooltip } from "@nextui-org/react";
+import { Button, Divider, Tooltip, Spinner } from "@nextui-org/react";
 import ProfilePictureAvatar from "@/components/common/ProfilePictureAvatar";
 import { useCollabContext } from "@/contexts/collab";
 import ChatMessage from "@/types/chat_message";
 import { Icons } from "@/components/common/Icons";
+import { OpenAI } from "langchain/llms/openai";
+import { ConversationChain } from "langchain/chains";
+import { BufferMemory, ChatMessageHistory } from "langchain/memory";
+import { HumanMessage, AIMessage } from "langchain/schema";
+import dotenv from "dotenv";
+
+dotenv.config();
```

Comment:
might not be a good idea to access local env here, as this is a client-side component, I doubt it can access .env values from when it gets loaded in client browser.

Reply:
oops i was supposed to to remove that part, forgot about it. thanks for the catch